Source/WebKit/GPUProcess/cocoa/GPUProcessCocoa.mm

```diff
@@ -89,6 +94,38 @@
 }
 #endif
 
+void GPUProcess::platformInitializeGPUProcess(GPUProcessCreationParameters& parameters)
+{
+#if PLATFORM(MAC)
+    auto launchServicesExtension = SandboxExtension::create(WTFMove(parameters.launchServicesExtensionHandle));
+    if (launchServicesExtension) {
+        bool ok = launchServicesExtension->consume();
+        ASSERT_UNUSED(ok, ok);
+    }
+
+    // It is important to check in with launch services before setting the process name.
+    launchServicesCheckIn();
+
+    // Update process name while holding the Launch Services sandbox extension
+    updateProcessName();
+
+    // Disable relaunch on login. This is also done from -[NSApplication init] by dispatching -[NSApplication disableRelaunchOnLogin] on a non-main thread.
+    // This will be in a race with the closing of the Launch Services connection, so call it synchronously here.
+    // The cost of calling this should be small, and it is not expected to have any impact on performance.
+    _LSSetApplicationInformationItem(kLSDefaultSessionID, _LSGetCurrentApplicationASN(), _kLSPersistenceSuppressRelaunchAtLoginKey, kCFBooleanTrue, nullptr);
```

Comment:
I don't think this is required in the GPU process, since it was a workaround for a sandbox issue when NSApplication is being used in the WebContent process.